src/cli.rs

```diff
@@ -252,6 +269,40 @@ mod tests {
     use crate::models::secondary_device::{MockDevice, MockDeviceFactory};
     use mockall::predicate::eq;
 
+    trait ExpectReadWrite {
```

Comment:
how is that different from `UserInterface` ?

Reply:
It's just a way to extend MocUserInterface (generated by automock) to simplify expectations :  "self.expect_read().times(1).returning(move || Ok(r.clone()));" 

It is not the implementation of UserInterface, it's just a wrapper around the automock `expect_read`, that handle `read` itself